exonum-light-client/src/main/java/com/exonum/client/HttpUrlHelper.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.net.URL;
+import java.util.Map;
+import okhttp3.HttpUrl;
+import okhttp3.HttpUrl.Builder;
+
+final class HttpUrlHelper {
+
+  static HttpUrl getFullUrl(URL host, String prefix, String relativeUrl,
+      Map<String, String> query) {
+    requireNonNull(query);
+    Builder builder = urlHostBuilder(host)
+        .addPathSegments(normalize(prefix))
+        .addPathSegments(normalize(relativeUrl));
+    query.forEach(builder::addEncodedQueryParameter);
+
+    return builder.build();
+  }
+
+  private static HttpUrl.Builder urlHostBuilder(URL host) {
+    requireNonNull(host);
+    HttpUrl.Builder builder = new HttpUrl.Builder()
+        .scheme(host.getProtocol())
+        .host(host.getHost());
+    if (host.getPort() != -1) {
+      builder.port(host.getPort());
+    }
+    return builder;
+  }
+
+  /**
+   * Removes heading slash from the path.
+   * Useful because underlying OkHttp applies slashes when constructing paths.
```

Comment:
Doesn't okhttp do that itself? Also, double slashes must not cause any problems, mustn't they?

Reply:
No, okhttp doesn't do that :( 
Urls with double stashes are valid. But such urls semantically identical but not equal. It may cause issues when storing them in collections and potentially cause some issues on a server side, for example caching (where the cache key is a path to the resource).
I see this question is more complex: such as we allow users to specify host and prefix in a string form may be do we need to have a full normalization urls support, for example as described [here](https://github.com/Norconex/commons-lang/blob/master/norconex-commons-lang/src/main/java/com/norconex/commons/lang/url/URLNormalizer.java#L56) ?

Reply:
I see, do I understand it correctly that we do the normalization to aid caching on a (possibly, intermediate) server in cases where the user has optional leading and/or trailing slashes? Isn't the naive approach used here enough?

Reply:
Yes, right.
But, I believe, we can simplify this and remove normalization and rely fully on OkHttp.

Reply:
I must admit that the URLs look better normalized, but agree it is not strictly needed (if a user cares, they can omit the slashes) :-)